cppsdk/include/playfab/PlayFabEvent.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#ifndef DISABLE_PLAYFABENTITY_API
+
+#include <playfab/PlayFabError.h>
+#include <playfab/PlayFabEventsDataModels.h>
+
+namespace PlayFab
+{
+    enum struct PlayFabEventServerProcessingType
+    {
+        Default,
+        Lightweight,
+        Heavyweight
+    };
+
+    /// <summary>
+    /// Interface for any event
+    /// </summary>
+    struct IPlayFabEvent
+    {
+    public:
+        virtual ~IPlayFabEvent() {}
+    };
+
+    /// <summary>
+    /// PlayFab-specific implementation of an event
+    /// </summary>
+    struct PlayFabEvent : public IPlayFabEvent
+    {
+    public:
+        void SetName(const std::string& eventName);
+        void SetProperty(const std::string& name, const std::string& value);
+        void SetProperty(const std::string& name, const bool value);
+        void SetProperty(const std::string& name, const int8_t value);
```

Comment:
How do we handle properties that already exist, ones that exist but are of a different type? Do we need to ever return false that we could not set it?

Reply:
It currently simply replaces, that's how Aria does it. We can change that 

---
In reply to: [227990835](https://github.com/PlayFab/XPlatCppSdk/pull/6#discussion_r227990835) [](ancestors = 227990835)